spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/ColumnarBatchReader.java

```diff
@@ -55,8 +55,14 @@ public ColumnarBatchReader(List<VectorizedReader<?>> readers) {
   @Override
   public void setRowGroupInfo(
       PageReadStore pageStore, Map<ColumnPath, ColumnChunkMetaData> metaData, long rowPosition) {
-    super.setRowGroupInfo(pageStore, metaData, rowPosition);
-    this.rowStartPosInBatch = rowPosition;
+    setRowGroupInfo(pageStore, metaData);
+  }
+
+  @Override
+  public void setRowGroupInfo(
+      PageReadStore pageStore, Map<ColumnPath, ColumnChunkMetaData> metaData) {
+    super.setRowGroupInfo(pageStore, metaData);
+    this.rowStartPosInBatch = pageStore.getRowIndexOffset().orElse(0L);
```

Comment:
if `pageStore.getRowIndexOffset()` is empty, does it mean [getRowIndexOffset()](https://github.com/apache/parquet-java/blob/master/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ColumnChunkPageReadStore.java#L379) returns a negative value? Shall we throw Exception instead of default it to 0?

Reply:
That is a good question.
As I understand it, the `PageReadStore` implementation (`ColumnChunkPageReadStore`) is normally constructed with the `rowIndexOffset`, but if the offset is not available then it is constructed with -1 for the `rowIndexOffset`. `PageReadStore::getRowIndexOffset()` will not return a negative value; it will return `Optional.empty()` in that case.
I suppose we can throw an IllegalArgumentException instead in such a situation, instead of setting `rowStartPosInBatch` to 0.
@flyrain do you have an opinion on this?
Is there someone who knows Parquet well who can confirm that in normal operation, `PageReadStore::getRowIndexOffset()` should *not* return `Optional.empty()`?